java/lib/src/main/java/truelayer/signing/Verifier.java

```diff
@@ -0,0 +1,216 @@
+package truelayer.signing;
+
+import com.nimbusds.jose.*;
+import com.nimbusds.jose.crypto.ECDSAVerifier;
+import com.nimbusds.jose.jwk.ECKey;
+
+import java.nio.charset.StandardCharsets;
+import java.security.interfaces.ECPrivateKey;
+import java.security.interfaces.ECPublicKey;
+import java.util.*;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Builder to verify a request against a `Tl-Signature` header using a public key.
+ */
+final public class Verifier {
+
+    private final ECPublicKey publicKey;
+
+    private String method = "";
```

Comment:
Method should have a default value (e.g. POST).

Reply:
Actually I don't think _verifier_ method should have a default, just _sign_.